2_program_structure/chess_board.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Write a program that creates a string that represents an 8×8 grid, using newline
+ * characters to separate lines. At each position of the grid there is either a space
+ * or a "#" character. The characters should form a chessboard.
+ *
+ * Passing this string to console.log should show something like this:
+ *
+ *  # # # #
+ * # # # #
+ *  # # # #
+ * # # # #
+ *  # # # #
+ * # # # #
+ *  # # # #
+ * # # # #
+ *
+ * When you have a program that generates this pattern, define a binding size
+ * = 8 and change the program so that it works for any size, outputting a grid
+ * of the given width and height.
+ */
+
+let board = '';
+let size = 8;
```

Comment:
Por ejemplo, digamos que quisiera mostar un board de 5x5, o uno de hasta 16x16, ¿qué cambios consideras que deberíamos hacer en el código para que funcione para cuálquier dimensión de board?

Reply:
Lo había dejado con tamaño de 8x8 dado que fue el especificado, pero si quisiera uno del tamaño que especifique el usuario en runtime, dependeria de donde se ejecute ese js file: si es en el browser, con un `prompt()` y convertirlo a number, pero en Node (en mi caso lo ejecuto y debuggeo asi jaja) como no hay `prompt()` built-in, se tendría que usar una librebria como prompt o readline a como investigué, y convertirlo a number como en la opcion anterior. Dado que la meta es hacer FE en un futuro, puedo cambiar el valor a que sea el indicado en un prompt, si te parece mejor.

Reply:
Me encanta que estés pensando en cómo interactuarían los usuarios finales con este programa <3

Lo único que comentaría es, que agregando la llamada a `prompt` al script, lo haces muy enfocado al browser. Entonces si quisieras aplicarlo en Node, tendrías que hacer malabares para saber si usar `prompt` o `readline`.

Por el momento, y es un consejo para aplicar en general, hay que matener nuestros programas/funcionalidad agnósticos de la UI/Front-end donde va a ser aplicado. A lo que quería llegar con el comentario orignial, era agrupar todo el script en una función que acepte `size` como parámetro y que regrese el board que construyes al final. Ahora que ya tienes la funcionalidad agnóstica del Front-end/Environment, ya luego puedes enfocarte en aplicar la función para el browser, o para node, o para cualquier environment que ejecute JavaScript :D

Reply:
Ohhh va, ya veo por donde iba tu comentario. Lo tendré siempre presente el buscar por lo pronto que la funcionalidad sea agnóstica, empezando con este ejercicio implementando lo que pediste. Thanks ✨